downstairs/src/region.rs

```diff
@@ -24,13 +23,66 @@ pub struct Extent {
 #[derive(Debug)]
 pub struct Inner {
     file: File,
-    meta: ExtentMeta,
+    metadb: Connection,
+}
+
+impl Inner {
+    fn flush_number(&self) -> Result<u64> {
+        let mut stmt = self
+            .metadb
+            .prepare("SELECT value FROM metadata where name='flush_number'")?;
+        let flush_number_iter = stmt.query_map([], |row| row.get(0))?;
+
+        let mut flush_number_values: Vec<u64> = vec![];
+        for flush_number_value in flush_number_iter {
+            flush_number_values.push(flush_number_value?);
+        }
+
+        assert!(flush_number_values.len() == 1);
+
+        Ok(flush_number_values[0])
+    }
+
+    fn set_new_flush_number(&self, new_flush: u64) -> Result<()> {
+        /*
+         * When we write out the new flush number, the dirty bit in the file
```

Comment:
Nit: Is this "the dirty bit in the record" now?  We don't have a file for each extent any longer.
I'm not sure what the proper database term is here, if it's not record.

Also, do we have TRUE and FALSE in database language, or do we just have to know that 0 is false
and 1 is true?

Reply:
> Also, do we have TRUE and FALSE in database language, or do we just have to know that 0 is false
and 1 is true?

On the subject of types in SQLite, it is indeed pretty loosey-goosey.  There is a [Datatypes in SQLite Version 3](https://www.sqlite.org/datatype3.html) document on their site that describes the situation in some detail.

In short, for booleans, `1` and `0` are indeed what it has available, and you're expected to convert them (however you do that) into the representation you need.

Reply:
Fixed comment in 24da54d, and yeah - SQLite is pretty "flexible" :)